### PR TITLE
Switch WebKitGTK testing to Ubuntu and add a nightly channel

### PR DESCRIPTION
**Pull request by clopez**
This PR adds two changes:

1) Switches testing with WebKitGTK to Ubuntu (removes the need for the extra dockerfile)

    Previously (#18595 and #19466) a Dockerfile based on Debian was added for WebKitGTK testing with the idea of reusing the built products that build.webkit.org is producing.
    
    That was done with the idea of having dozens of new nightly builds available at a day, which doesn't seem to match the needs for WPT testing/runs. It seems that having one new nightly a day (or even two a week) is more than enough. At that rate is more than reasonable to do new builds for testing with Ubuntu 18.04.
    
    And it seems that harmonizing the testing of the Linux based browsers on the same docker image has several benefits, including making easier future maintenance.
    
2) Adds a nightly channel for testing with WebKitGTK on taskcluster.

    The nighly channel downloads the last nightly tarball available at https://webkitgtk.org/built-products and installs it on taskcluster when the test start.
    
    This tarball is generated using the webkitgtk internal JHBuild, that builds several third-party libraries needed for webkitgtk layout tests and then builds webkitgtk on top of this libraries.
Because of this, to use this tarball it is required to installing quite a lot of extra dependencies (that are needed by this extra third-party libraries).

    A script is included inside the tarball to install this dependencies. This script is executed at test time in task cluster. But because installing them take quite a bit of time (several minutes), the dockerfile is modified to have them already installed and speed up this process. Having them already installed on the docker image is not strictly necessary, it just speed up the testing process.

//cc @jgraham @foolip

### CODE REVIEW

**Comment by clopez** [comment]
Example of wpt.fyi results with the new setup:

* stable [`2.26.1`]: https://staging.wpt.fyi/results/?run_id=255670028
* nightly [`2.27.1 (r250843)`]: https://staging.wpt.fyi/results/?run_id=235700040

File: tools/docker/Dockerfile
Comment (foolip) [inline tools/docker/Dockerfile]: Is the upgrade bit important here? Seems like a sensible thing to do, but isn't the base ubuntu:18.04 updated regularly with the most recent packages?

File: tools/docker/Dockerfile
Comment (foolip) [inline tools/docker/Dockerfile]: How much size does this add to the docker image, and how much time does it save?

Since this would only take effect when updating the image, which we don't do often, will the dependencies not become increasingly out of sync and need to be updated at runtime?

**Review by foolip — COMMENTED** [review]
This looks good modulo the question about installing deps in the image, and I'd like @jgraham to review since this will involve another weekly run if my suggestion for stable/nightly cadence is accepted.

File: .taskcluster.yml
Comment (clopez) [reply .taskcluster.yml]: Indeed, It will be nice if it can be bumped. But I was planning to do that after this lands because I can only have a docker image uploaded and if I upload a new web-platform-tests:0.34 now then the previous based on debian that is running now would be deleted.

Even if the image is not bumped, nightly builds can run ok. The only difference is that they take more time at test time because the dependencies have to be installed at test time for each taskcluster run.

File: tools/docker/Dockerfile
Comment (clopez) [reply tools/docker/Dockerfile]: I really don't know if its upgraded regularly, but I suspect is not upgraded as much as it should... because when you run that upgrade, it actually upgrades packages.

File: tools/docker/Dockerfile
Comment (clopez) [reply tools/docker/Dockerfile]: It makes the docker image 1GB bigger (from 1.27GB to 2.2GB). It saves around 6-10 minutes:
 * 3-7 minutes of download (depends on the download speed of the ubuntu apt repositories)
 * ~3 minutes of installing packages

File: tools/ci/run_tc.py
Comment (clopez) [reply tools/ci/run_tc.py]: The file is around 150MB. I guess its ok, but is still big.
In any case I think we can use `urlopen` to write the file to disk in chunks. I will try.

Comment (foolip) [reply tools/ci/run_tc.py]: Looks like it's pretty straightforward: https://stackoverflow.com/a/1517728

File: .taskcluster.yml
Comment (clopez) [reply .taskcluster.yml]: Yes, that makes sense.. but I'm not sure how/where the frequency of the runs is configured? Or do you mean to add `webkitgtk_minibrowser:nightly` to `refs/heads/epochs/daily` ?

File: tools/docker/Dockerfile
Comment (foolip) [reply tools/docker/Dockerfile]: Given that WebKitGTK is still in a bit of an experimental phase that sounds like it's adding a bit too much burden on the other uses of the same image and could increase total transfer size, but I'll let @jgraham weigh in on that.

File: tools/docker/Dockerfile
Comment (foolip) [reply tools/docker/Dockerfile]: Alright, a bit of a drive-by, but that seems fine.

File: .taskcluster.yml
Comment (foolip) [reply .taskcluster.yml]: Yes, and move stable to weekly.

File: .taskcluster.yml
Comment (foolip) [reply .taskcluster.yml]: I think landing Dockerfile changes with image bumps at the same time makes the most sense, since otherwise there's a time when the Dockerfile in the tree doesn't reflect what's actually being used, and one can't figure out from a version like 0.33 or 0.34 from what it was built.

Comment (clopez) [reply .taskcluster.yml]: That is a good idea. I just bump the reference and land the PR myself to ensure everything is synced. Thanks for the tip.

File: .taskcluster.yml
Comment (clopez) [reply .taskcluster.yml]: I tried pushed a commit on ffea8524ab62836835fbaf48a8bab0bc53cc088c doing that.

Not sure if the triggers ```refs/heads/triggers/webkitgtk_minibrowser_stable``` and ```refs/heads/triggers/webkitgtk_minibrowser_nightly``` should stay? (I deleted them)

File: tools/ci/run_tc.py
Comment (clopez) [reply tools/ci/run_tc.py]: Done in 68500be9108d77d3136682c57e2a23a717274973

File: tools/docker/Dockerfile
Comment (clopez) [reply tools/docker/Dockerfile]: Regarding how often they are updated: https://anchore.com/look-often-docker-images-updated/ (once or twice a month)

File: .taskcluster.yml
Comment (clopez) [reply .taskcluster.yml]: Bumped in 33240b88f8c5e17f89aedfca34eb2c387d2fb5b3

File: tools/ci/run_tc.py
Comment (foolip) [inline tools/ci/run_tc.py]: I'm not sure if /tmp/ is always writable, but would it make sense to use https://docs.python.org/2/library/tempfile.html#tempfile.NamedTemporaryFile to not have to worry about it?

Comment (clopez) [reply tools/ci/run_tc.py]: AFAIK `/tmp` is assumed to be always writable. Actually `tempfile.NamedTemporaryFile` returns a file on `/tmp`. But now that you mention it, I see that I'm not deleting the temporal file on `/tmp` after unpacking it. And I can have python do that automatically when the file is closed by just using `tempfile.NamedTemporaryFile`. Let's try it.

Comment (clopez) [reply tools/ci/run_tc.py]: Pushed d601996c19ab6afced2eb82179d97313fbd20a95 with the ```tempfile.NamedTemporaryFile``` approach

File: .taskcluster.yml
Comment (foolip) [reply .taskcluster.yml]: Removing them doesn't break anything, I think, but I've left the other ones pointing to the last thing that was triggered manually.

Comment (foolip) [reply .taskcluster.yml]: It's also nice to have them to make it more likely that two people trying to trigger manually at the same time notice that, as documented in https://web-platform-tests.org/running-tests/from-ci.html.

**Review by foolip — APPROVED** [review]

File: tools/docker/Dockerfile
Comment (jgraham) [reply tools/docker/Dockerfile]: Yeah, I think I'd prefer to make the WebKitGtk runs bear that cost for now. If we think that adding 10 minutes per job is too much I guess we could have a second Docker image here which inherits from the base one and just adds the extra bits to do this update (yes, that undoes part of the motivation for this PR). That adds the maintainance burden of requiring us to update both dockerfiles when we change the base image.

**Review by jgraham — CHANGES_REQUESTED** [review]

**Comment by clopez** [comment]
Right. Let's try to keep this simple for now and avoid bloating the image with this extra deps. I think the 10 minutes of extra time is something that we can live for now, since it runs daily.
If in the future there is timeouts or we want to speed up webkitgtk runs we can rethink this.
I will rebase the patches cleaned without changes to the dockerfile.

File: tools/docker/Dockerfile
Comment (foolip) [reply tools/docker/Dockerfile]: @clopez I see you've rebased now but these changes remain. Please ping when it's ready for final review.

**Comment by clopez** [comment]
Rebased and cleaned patches. @jgraham does it look ok now?

**Review by jgraham — APPROVED** [review]

**Comment by foolip** [comment]
@clopez looks like the daily build failed both times since this was landed:
https://tools.taskcluster.net/groups/c1CPobZVRQuTrZpqDRoyUw (for 20a55f606de85d95a46cd19b423c9215cb39898c)
https://tools.taskcluster.net/groups/ABUsWSTDSxymeh5oR1MWBQ (for 50fa6ade9bda8c138ee37878c8442f33af09cb7d)

It's wpt-webkitgtk_minibrowser-nightly-testharness-4 that timed out after 2 hours in both cases. It wasn't at the same point, not sure if things had hung or if it was still running. Can you investigate?

https://tools.taskcluster.net/groups/IZIeBnWZQg2hmBkDgIMT5Q (for 823aecebaa2c04fd357c017849ed618107f9fa1b) is the last daily run before these changes, which succeeded.

**Comment by clopez** [comment]
I think the timeout is caused because of this extra 10-15 minutes needed for the download of dependencies. I checked this with @jgraham and he suggests to first try to raise the number of chunks before trying to raise the timeout value (is set a 2 hours). I opened a PR for that in #19724 
I also observed on some runs I have been trying on the triggers/webkitgtk_minibrowser_nightly branch that sometimes the download aborts. I opened a PR for that in #19725

**Pull request merged**
